### ISSUE TITLE
Add new grid functions max() and min()

### ISSUE THREAD

**Issue by jbusecke** [xgcm/xgcm#90]
I am trying to approximate advective fluxes for the MOM5 ocean model offline. 
xgcm is well suited to perform the interpolation of horizontal fluxes from velocity points (B-grid; upper right corner) to the cell faces.

Some operations, however require to calculate the minimum of a variable (in this case mass
per unit area; see [MOM5 manual](http://www.mom-ocean.science/web/docs/project/MOM5_manual.pdf) section 14.5.1 and 14.5.2).

@rabernat would it be hard to add two new functions that implement a min and max function?
It should basically work exactly like the diff and interp functions (in terms of the grid placement), just the value should be either the min or max of the neighboring pairs.

I would be happy to work on a PR, just wanted to see if you anticipate any problems or had any tips.\

Edit: In particular I wanted to wait and ask if this would interfer with #88.

**Comment by jbusecke** [comment]
While at this, another useful function would be a very simple 'grid.shift' function.

The purpose of this would be to fix netcdffiles that did not assign the correct grid position.

I found some files recently that had fluxes on the grid face saved on the tracer point and I have been using this to reassign the cell face positions

```
def grid_shift(da, axis, grid):
    ref = grid.interp(da, axis)
    return xr.DataArray(da.data, coords=ref.coords, dims=ref.coords)
```

I think it would be useful (and probably straightforward to implement) to achieve this from within xgcm?

**Comment by jbusecke** [comment]
I was playing around with the min and max function and have a crude working implementation. I am still running into trouble at coordinate discontinuities, but I am sure we can handle that with some of the 'boundary_discontinuity' logic I wrote for the grid_generation module. While digging there, I was also hoping to look into #92. These seem related.

I will need this functionality in the near time, and just wanted to gauge if there is interest in a PR at all. Or if this is regarded as something to be handled outside of xgcm.

**Comment by rabernat** [comment]
Please submit a PR!

The way the code is structured, it should be very straightforward to add any "binary function" such as min, max, etc. I support this 100%

**Comment by jbusecke** [comment]
Yeah exactly! I added clones of `axis.diff()` and `grid.diff()` that execute `xr.ufuncs.minumum(left,right)` instead of `left-right` in the return line. Let me see what I can do about the discontinuity. 

Should I include the shift function too? 
This would be even easier: Just return the left or right value and that's it. 

This will be cool to have and I could develop a pangeo use case for some CM2.6 budget reconstructions down the line :-).

**Comment by rabernat** [comment]
Feel free to put all of these into a PR. We can discuss the implementation more there.

We will try to find the most efficient way of wrapping such functions that involves as little boilerplate code as possible. For example, see how dask wraps numpy array functions:
https://github.com/dask/dask/blob/master/dask/array/routines.py#L171

**Comment by jbusecke** [comment]
closed via #96
